core/src/avm2/globals/flash/display/bitmap_data.rs

```diff
@@ -1369,3 +1374,73 @@ pub fn compare<'gc>(
         None => Ok(EQUIVALENT.into()),
     }
 }
+
+/// Implements `BitmapData.pixelDissolve`.
+pub fn pixel_dissolve<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(bitmap_data) = this.and_then(|t| t.as_bitmap_data()) {
+        bitmap_data.check_valid(activation)?;
+
+        let src_bitmap_data = args.get_object(activation, 0, "sourceBitmapData")?;
+
+        let source_rect = args.get_object(activation, 1, "sourceRect")?;
+
+        let src_min_x = source_rect
+            .get_public_property("x", activation)?
+            .coerce_to_i32(activation)?;
+        let src_min_y = source_rect
+            .get_public_property("y", activation)?
+            .coerce_to_i32(activation)?;
+        let src_width = source_rect
+            .get_public_property("width", activation)?
+            .coerce_to_i32(activation)?;
+        let src_height = source_rect
+            .get_public_property("height", activation)?
+            .coerce_to_i32(activation)?;
+
+        let dest_point = args.get_object(activation, 2, "destPoint")?;
+        let dest_point = (
+            dest_point
+                .get_public_property("x", activation)?
+                .coerce_to_i32(activation)?,
+            dest_point
+                .get_public_property("y", activation)?
+                .coerce_to_i32(activation)?,
+        );
+
+        let random_seed = args.get_i32(activation, 3)?;
+
+        let num_pixels = args.get_i32(activation, 4)?;
+        if num_pixels < 0 {
+            return Err(Error::AvmError(range_error(
+                activation,
+                &format!("Parameter numPixels must be a non-negative number; got {num_pixels}."),
```

Comment:
Nit: I expect this should be prefixed with `Error #2027: `?

Reply:
I think I added that at some point, but then I looked at other code like https://github.com/ruffle-rs/ruffle/blob/master/core/src/avm2/globals/flash/display/display_object_container.rs#L60-L64 , and I got in doubt about whether to keep it. I can see that for some of the `AvmError()`, the error code is include in the message, and in others of them, the error code is not included. Should I include it? I think that I will make minor changes to the tests if I include it, since then I can include the `e:Error` in the message instead of printing a custom message.

Reply:
Because I think the official Flash Player does include the number in the string version of the error that is thrown.

Reply:
I think originally we somewhat assumed the error code to be part of tostring, but we discovered that wasn't the case. Places like that should be updated to include it.

Strictly speaking we don't *need* to be 1:1 with Flash error messages, and in some cases we deliberately add more info or give more correct messages than sometimes outright wrong ones used by Flash. However, error codes are useful for humans I guess!

Reply:
I will add it to the code and to the tests.

Reply:
I have added it :) .